amftrack/pipeline/launching/run.py

```diff
@@ -0,0 +1,113 @@
+import os
+import pandas as pd
+from amftrack.util.sys import (
+    path_code,
+    temp_path,
+    update_plate_info,
+    get_current_folders,
+)
+from typing import List
+
+from time import time_ns
+from tqdm.autonotebook import tqdm
+import subprocess
+
+
+def make_stitching_loop(directory: str, dirname: str, op_id: int) -> None:
+    """
+    For the acquisition directory `directory`, prepares the stitching script used to stitch the image.
+    """
+    # TODO(FK): handle the case where there is no stiching_loops directory
+    a_file = open(
+        os.path.join(path_code, "pipeline/scripts/stitching_loops/stitching_loop.ijm"),
+        "r",
+    )
+
+    list_of_lines = a_file.readlines()
+    list_of_lines[4] = f"mainDirectory = \u0022{directory}\u0022 ;\n"
+    list_of_lines[29] = f"\t if(startsWith(list[i],\u0022{dirname}\u0022)) \u007b\n"
+
+    file_name = f"{temp_path}/stitching_loops/stitching_loop{op_id}.ijm"
+    a_file = open(file_name, "w")
+    a_file.writelines(list_of_lines)
+    a_file.close()
+
+
+def run_stitch(directory: str, folders: pd.DataFrame) -> None:
+    """
+    Runs the stiching loop.
+    :param directory: is the folder containing the directories for each acquisition
+    :param folder: is a pandas dataframe with information on the folder
+    """
+    # TODO(FK): use a str not a pandas dataframe
```

Comment:
Why use a str not a pandas dataframe?

Reply:
C'est que la dataframe contenant les infos, c'est un objet assez compliqué à créer, et qui me paraissait pas nécessaire de connaitre pour certaines fonctions (pour stich, seul le nom du dossier est important par exemple).
En gros la fonction stich, tu pourrais juste l'appeler sur un `path`. Alors que là pour l'appeler, il faut utiliser toute la machinerie de update le data_info.json, set up le directory, load la data frame générale et tout.
Cela dit, maintenant qu'on peut récupérer la dataframe `folders` sans l'appel à dropbox, ça devient plus facile de créer l'objet `folders` et avec les test on peut facilement en créer un en cas de test. 
Donc la modif est vraiment pas une urgence. Et dans tous les cas je laisserai la fonction run_stich, c'est plutôt encapsuler la partie de la fonction qui fait le stiching dans une fonction qui prend juste un path str en argument pour les cas simple où on veut pas créer l'objet dataframe

Reply:
Okay oui l'encapsulation parait pertinente.
Aussi parce qu'il y a une redondance entre beaucoup de fonction pour extraire du dataframe `folders ` la liste de nom de folders à traiter. Donc il y aurait des choses à factoriser En fait c'est agréable d'avoir une fonction qui prend `folders `en entier parce que c'est facile de checker avant de lancer l'opération quel est l'état des folders sur lesquels ont veut la lancer. Mais en général les seules colonnes nécessaire sont la colonne "Plate" et la colonne "folder".

Reply:
Dacc, bah je vais laisser un TODO "a factoriser" alors